### PR TITLE
Overflow protection

### PR DESCRIPTION
**Pull request by dwalintukan**

### CODE REVIEW

File: contracts/BodhiToken.sol [hunk 1]
```diff
@@ -81,6 +81,8 @@ contract BodhiToken is StandardToken, Ownable {
   }
 
   function mintReservedTokens(uint256 _amount) onlyOwner {
+    require(totalSupply + _amount > totalSupply);
```
Comment (TT1943) [inline contracts/BodhiToken.sol]: what does this mean? _amount > 0?
If you mean to check totalTotalSupply, I think
line 85's `require(checkedSupply <= tokenTotalSupply);` already achieve what you are trying to do

File: contracts/BodhiToken.sol [hunk 2]
```diff
@@ -96,8 +98,11 @@ contract BodhiToken is StandardToken, Ownable {
    * @dev Function to mint tokens
    */
   function mint(address _to, uint256 _amount) internal returns (bool) {
+    require(balances[_to] + _amount > balances[_to]);
```
Comment (TT1943) [inline contracts/BodhiToken.sol]: do you mean _amount > 0?

File: contracts/BodhiToken.sol (hunk 1 above)
Comment (dwalintukan) [reply contracts/BodhiToken.sol]: This is to check for overflow: https://github.com/ConsenSys/smart-contract-best-practices#integer-overflow-and-underflow

Even though unlikely in this case, its important to have the check. any uint can reset back to 0 if it hits the maximum value.

File: contracts/BodhiToken.sol (hunk 2 above)
Comment (dwalintukan) [reply contracts/BodhiToken.sol]: nope, see comment above about overflow.

**Pull request closed without merging**